File: server/webflix/src/main/java/com/instil/webflix/basket/service/BasketService.java
package com.instil.webflix.basket.service;

import com.instil.webflix.basket.model.BasketSummary;
import com.instil.webflix.movies.model.Movie;
import com.instil.webflix.security.model.Account;

public interface BasketService {
	int getItemCount(Account account);

	void addMovieToBasket(Account account, Movie movie);

	void clearBasket(Account account);

	BasketSummary getSummary(Account account);

    void clearMovie(Account account, Long movieId);

    void checkout(Account account);
}

File: client/webflix/src/app/service/basket/basket.service.ts
import {Observable} from 'rxjs/Observable';
import {Movie} from '../../model/movie';
import {BasketSummary} from '../../model/basket-summary';

export abstract class BasketService {
  abstract getBasketItemCount(): Observable<number>;

  abstract addToBasket(movie: Movie): Observable<boolean>;

  abstract get basketCount(): Observable<number>;

  abstract getBasketSummary(): Observable<BasketSummary>;

  abstract clearBasket(): Observable<boolean>;

  abstract removeMovie(movie: Movie): Observable<boolean>

  abstract checkout(): Observable<boolean>
}
